src/ui/tui.py

```diff
@@ -1,88 +1,136 @@
-        else:
-            print('Absolut kritisch mein Freund')
-
-    def login_prompt(self) -> Optional[int]:
-        # idgaf about this method, get rekt
-        str(input('Username: '))
-        int(input('Account ID: '))
-        str(input('Passwort: '))
-        conUser = True
-        return conUser
-
-    def choose_option(self) -> Optional[int]:
-        options = [
-            "Einzahlung",
-            "Auszahlung",
-            "Überweisung",
-            "Kontostand"
-        ]
-        print('Wähle aus:')
-        [print(f"{index + 1} = {option}") for index, option in enumerate(options)]
-        try:
-            option = int(input())
-            assert option > 0, "Gültig oder ich stech dich ab"
-            chosen = options[option - 1]
-            return chosen
-        except TypeError:
-            print("Bitte geben sie einen gültigen Integer ein, danke Ronneberg")
-            return False
-        except IndexError:
-            print("Ist keine gültige Option")
-            return False
-
-    def change_user_or_stay_logged_in(self) -> Optional[int]:
-        try:
-            option = int(
-                input('Weiter Aktionen mit diesen Account tätigen?\n1 = Nein | 2 = Ja\n')
-            )
-            return option - 1
-        except TypeError:
-            print("Bitte geben sie einen gültigen Integer ein, danke Ronneberg")
-            return False
-
-    def keep_running(self) -> Optional[int]:
-        try:
-            option = int(
-                input('Programm beenden?\n1 = Nein | 2 = Ja\n')
+            storage_type = input(
+                "Was willst du haben? JSON oder SQL? "
             )
-            return option - 1
-        except TypeError:
-            print("Programm wird fortgefahren")
-            return False
 
+            try:
+                storage = storage_types[
+                    storage_type.lower()
+                ]()
+                self.storage = storage
+            except KeyError:
+                print("Falsche Eingabe!")
+            else:
+                break
+
+        self.user = self.login_user()
+
+        action = self.select_action()
+        action()
+
+    def create_account(self) -> Kunde:
+        # Still needs some validation
+        # Check if User has account
+        # if not he needs to input credentials
+        new_account = input("Haben Sie bereits einen Account J/N?: ")
+        if new_account.lower() == "j":
+            while True:
+                kundendaten= {}
+                kundendaten["username"] = input("Username: ")
+                kundendaten["password"] = getpass("Passwort: ")
+                kundendaten["name"] = input("Vor und Nachname: ")
+                kundendaten["strasse"] = input("Straße und Hausnummer: ")
+                kundendaten["stadt"] = input("Stadt: ")
+                kundendaten["plz"] = input("Postleitzahl: ")
+                kundendaten["geb_date"] = input(datetime.date())
```

Comment:
Das macht keinen Sinn.
plz muss ein Integer sein (input erzeugt nur strings by default) und die Zeile darunter ist komplett sinnfrei, dort muss einmal dd:mm:yyyy in den datetime.date Konstruktor eingefügt werden.

Außerdem ist der import falsch, es muss entweder ``from datetime import date`` sein und dann statt ``datetime.date()`` nur ``date()`` sein, oder der Import wird auf ``import datetime`` geändert.